src/LanguageServer/ProtocolUnitTests/Diagnostics/AdditionalFileDiagnosticsTests.cs

```diff
@@ -63,10 +63,10 @@ public async Task TestWorkspaceDiagnosticsWithRemovedAdditionalFile(bool useVSDi
         var results = await RunGetWorkspacePullDiagnosticsAsync(testLspServer, useVSDiagnostics);
         Assert.Equal(3, results.Length);
 
-        Assert.Empty(results[0].Diagnostics);
+        AssertEx.Empty(results[0].Diagnostics);
```

Comment:
curious why Assert.Empty doesn't wor khere? (or in all the other places changed).

Reply:
@CyrusNajmabadi tried to summarize a lot of the reasons like this in the PR description + doc comments on the methods. In this case it's because xUnit NRT annotated their APIs and `Empty`, `Single`, etc ... do not accept `null`. Given that diagnostics assigns different meaning to `null` and `Empty` I didn't want to make broad changes to the test implementation. Instead I added null annotated versions of the methods to `AssertEx` and redirected the impacted sites to those.